File: 1843d_at.py
from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    t = int(input())
    for _ in range(t):
        n = int(input())
        tree = [[] for i in range(n)]
        for i in range(n-1):
            u, v = map(int, input().split())
            u -= 1
            v -= 1
            tree[u].append(v)
            tree[v].append(u)
        leaf_count = [0 for _ in range(n)]
        stack = [(0, 0, -1)]
        while stack:
            node, visit_count, parent = stack.pop()
            if visit_count == 0:
                stack.append((node, 1, parent))
                for child in tree[node]:
                    if child != parent:
                        stack.append((child, 0, node))
            else:
                if tree[node] == [parent]:
                    leaf_count[node] = 1
                else:
                    leaf_count[node] = sum(leaf_count[child]
                                           for child in tree[node] if child != parent)
        q = int(input())
        for i in range(q):
            x, y = map(int, input().split())
            x -= 1
            y -= 1
            print(leaf_count[x] * leaf_count[y])


if __name__ == "__main__":
    main()

File: 1843d_at.js
"use strict";
const readline = require("readline");
let rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout,
    terminal: false,
});
let lines = [];
rl.on("line", (line) => {
    lines.push(line);
});
rl.on("close", () => {
    let t = parseInt(input());
    let tree, leaf_count;
    for (let i = 0; i < t; i++) {
        let n = parseInt(input());
        tree = new Array(n).fill(0).map(() => []);
        for (let j = 0; j < n - 1; j++) {
            let [u, v] = input().split(" ").map(Number);
            tree[--u].push(--v);
            tree[v].push(u);
        }
        leaf_count = Array(n).fill(0);
        let stack = [[0, 0, -1]];
        while (stack.length) {
            let [node, visit_count, parent] = stack.pop();
            if (visit_count === 0) {
                stack.push([node, 1, parent]);
                for (let child of tree[node]) {
                    if (child !== parent) {
                        stack.push([child, 0, node]);
                    }
                }
            } else {
                if (tree[node].length === 1 && tree[node][0] === parent) {
                    leaf_count[node] = 1;
                } else {
                    for (let child of tree[node]) {
                        if (child !== parent) {
                            leaf_count[node] += leaf_count[child];
                        }
                    }
                }
            }
        }
        let q = parseInt(input());
        for (let j = 0; j < q; j++) {
            let [x, y] = input().split(" ").map(Number);
            console.log(leaf_count[--x] * leaf_count[--y]);
        }
    }
    rl.close();
});

let currentLine = 0;
function input() {
    return lines[currentLine++];
}
